http/rest-client-reactive/src/main/java/io/quarkus/ts/http/restclient/reactive/ResourceAndSubResourcesClient.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkus.ts.http.restclient.reactive;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.eclipse.microprofile.rest.client.annotation.RegisterClientHeaders;
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+@RegisterRestClient
+@RegisterClientHeaders
+@Consumes(MediaType.TEXT_PLAIN)
+@Produces(MediaType.TEXT_PLAIN)
+public interface ResourceAndSubResourcesClient {
+    @Path("clients")
+    ClientsResource clients();
+
+    interface ClientsResource {
+        @Path("{id}")
+        ClientResource get(@PathParam("id") String id);
+    }
+
+    interface ClientResource {
+        @Path("/resource-server")
+        SubResource sub();
+    }
+
+    interface leafResource {
```

Comment:
Unless it is absolutely required, we should name classes and interfaces according to Java code conventions[1]. If it **is** absolutely required, please provide an explanation in code comments

[1] https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html